protocol/src/lib/uploads.config.ts

```diff
@@ -0,0 +1,209 @@
+/**
+ * Shared Uploads Configuration & Validation
+ *
+ * Pure constants, types, and validation logic consumed by both backend and frontend.
+ * All validation functions work with primitive types to avoid runtime dependencies.
+ */
+
+// File size limits in bytes
+export const FILE_SIZE_LIMITS = {
+  GENERAL: 10 * 1024 * 1024, // 10MB for general files
+  AVATAR: 4 * 1024 * 1024,   // 4MB for avatars
+} as const;
+
+// Maximum number of files per upload request
+export const MAX_FILES_PER_UPLOAD = 10 as const;
+
+// Supported file types based on Unstructured.io capabilities
+export const SUPPORTED_FILE_TYPES = {
+  // Document formats
+  DOCUMENTS: {
+    extensions: ['.pdf', '.doc', '.docx', '.txt', '.csv', '.xls', '.xlsx', '.json', '.md', '.ppt', '.pptx', '.rtf', '.odt', '.xml', '.yaml', '.yml', '.html', '.htm', '.epub'],
+    mimeTypes: [
+      'application/pdf',
+      'application/msword',
+      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+      'text/plain',
+      'text/csv',
+      'application/vnd.ms-excel',
+      'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+      'application/json',
+      'text/markdown',
+      'application/vnd.ms-powerpoint',
+      'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+      'application/rtf',
+      'application/vnd.oasis.opendocument.text',
+      'application/xml',
+      'text/xml',
+      'application/x-yaml',
+      'text/yaml',
+      'text/html',
+      'application/epub+zip'
+    ]
+  },
+  
+  // Image formats (for avatars)
+  IMAGES: {
+    extensions: ['.jpg', '.jpeg', '.png', '.gif', '.webp', '.bmp', '.tiff', '.tif', '.heic'],
+    mimeTypes: [
+      'image/jpeg',
+      'image/png',
+      'image/gif',
+      'image/webp',
+      'image/bmp',
+      'image/tiff',
+      'image/heic'
+    ]
+  },
+  
+  // Email formats
+  EMAIL: {
+    extensions: ['.eml', '.msg', '.mbox'],
+    mimeTypes: [
+      'message/rfc822',
+      'application/vnd.ms-outlook',
+      'application/mbox'
+    ]
+  },
+  
+  // Archive formats (read-only, no processing)
+  ARCHIVES: {
+    extensions: ['.zip'],
+    mimeTypes: [
+      'application/zip',
+      'application/x-zip-compressed',
+      'application/x-zip'
+    ]
+  }
+} as const;
+
```

Comment:
We don't support these. Supported ones are: csv, epub, html, md, pdf, txt, ppt, pptx, rtf, xls, xlsx, tsv, doc, docx, xml, json

Check unstructured docs for mime type validation.